### DISCUSSION TITLE
25. October 2022

### DISCUSSION THREAD

**Discussion by dimakuv** [gramineproject/gramine#985] [Meeting notes]
### Agenda

*(please write your proposed agenda items in comments under this discussion)*

- [ ] ~Dmitrii and Woju: Use TOML Python package version 0.10.2 (https://github.com/gramineproject/gramine/issues/990).~
- [ ] ~Benny and Deb: Current status of documentation restructuring.~ UPDATE: Moved to next meeting.
- [x] Mona: Discuss EOL users@gramineproject.io and moving to Discussions or some other option.
- [x] Michal: Discuss dropping OOT driver support in packages (but keep in source for developers and CI).

### Discuss EOL users@gramineproject.io and moving to Discussions

*Michal*: Why do we need to remove users@gramineproject.io email?

*Mona*: we have this email in README, [in "Getting help"](https://github.com/gramineproject/gramine/#getting-help). We should change to "GitHub Discussions".
- Also change "for bug reports" -> "for bug reports and feature requests".
- Also move users@gramineproject.io to the third line, so that users start with Gitter/Discussions/Issues.

~**TODO:** Create a PR!~ PR created: https://github.com/gramineproject/gramine/pull/1011.

### Discuss dropping OOT/DCAP driver support

*Woju*: proposal for from-source builds -- keep only two Gramine-build options:
1. OOT driver (only option for non-FLC machines, so must keep it)
2. in-kernel driver
  
*Woju*: proposal for Gramine packaging -- ship only a single package `gramine` (do **not** ship additional `gramine-dcap` and `gramine-oot`). This will work on both DCAP and in-kernel drivers. For OOT driver, people must compile themselves.

*Michal and Woju*: we are fine with the fact that the in-kernel SGX driver has "unofficial" compatibility with DCAP 1.6+. That's why using the same `gramine` package for (any version of) DCAP and in-kernel drivers works.

*Michal and Woju*: During Gramine-SGX boot, if `/dev/sgx_enclave` doesn't exist, then fallback to `/dev/sgx/enclave`. **TODO**: Create a PR with this fallback.

*Scott*: It's fine to remove support for OOT (in packages). This is what Scott is pushing for in all SGX projects.

*Woju*: Important note: the new packages (v1.4) will appear in January/February 2023 anyway, so there are still several months for people to move off the OOT driver in packages.

### Split of `common` dependency in Gramine codebase

*Borys*: Why do we need to split deps into three, as was done in early revisions of https://github.com/gramineproject/gramine/pull/1002?
- "Libc or not libc" is a wrong distinction. It should be just "gramine-specific" and "non-gramine-specific".
- ASan sources should be in "gramine-specific" dep.

*Borys*: we need to split `api.h` into two: "gramine-specific" and "non-gramine-specific".
  - No conclusion, need more people to review PR #1002 and decide on a split of `api.h` (if any).

**Comment by mkow** [comment]
- [ ] Discuss dropping OOT driver support in packages (but keep in source for developers and CI)

**Reply by woju** [reply]
and discuss removing `-Dsgx_driver=dcap*` (#997)

**Comment by monavij** [comment]
Discuss EOL [users@gramineproject.io](mailto:users@gramineproject.io) and moving to discussions or some other option.
